test/unit/formats/odata.js

```diff
@@ -930,6 +930,92 @@ describe('odata message composition', () => {
         });
       });
 
+      describe('with $skiptoken', () => {
+        const billy = { __id: 'cf9a1b5cc83c6d6270c1eb98860d294eac5d526d', age: 4, name: 'Billy' };
+        const blain = { __id: 'c76d0ccc6d5da236be7b93b985a80413d2e3e172', age: 6, name: 'Blaine' };
+
+        [
+          {
+            $top: 0,
+            skiptoken: { repeatId: billy.__id },
+            expectedNext: false,
+            expectedValue: [],
+          },
+          {
+            $top: 1,
+            skiptoken: { repeatId: billy.__id },
+            expectedNext: billy.__id,
+            expectedValue: [ billy ],
+          },
+          {
+            $top: 2,
+            skiptoken: { repeatId: billy.__id },
+            expectedNext: false,
+            expectedValue: [ billy, blain ],
+          },
+          {
+            $top: undefined,
+            skiptoken: { repeatId: billy.__id },
+            expectedNext: false,
+            expectedValue: [ billy, blain ],
+          },
+
+          {
+            $top: 0,
+            skiptoken: { repeatId: blain.__id },
+            expectedNext: false,
+            expectedValue: [],
+          },
+          {
+            $top: 1,
+            skiptoken: { repeatId: blain.__id },
+            expectedNext: false,
+            expectedValue: [ blain ],
```

Comment:
It looks like this result is inconsistent with a similar request for `repeatId: billy` above - here there is no `expectedNext`, whereas above the nextLink returned is a circular link to the original query.